java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -319,6 +322,17 @@ public CompletableFuture<Long> scard(String key) {
         return commandManager.submitNewCommand(SCard, new String[] {key}, this::handleLongResponse);
     }
 
+    @Override
+    public CompletableFuture<Long> clientId() {
```

Comment:
this doesn't need to go in BaseClient.  
`clientId` without a route is not available in the cluster client.